src/test/java/de/tum/in/www1/artemis/ModelingExerciseIntegrationTest.java

```diff
@@ -264,6 +268,28 @@ public void importModelingExerciseFromCourseToCourse() throws Exception {
         request.postWithResponseBody("/api/modeling-exercises/import/" + modelingExercise.getId(), modelingExercise, ModelingExercise.class, HttpStatus.CREATED);
     }
 
+    @Test
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void importModelingExerciseWithExampleSubmissionFromCourseToCourse() throws Exception {
+        var now = ZonedDateTime.now();
+        Course course1 = database.addEmptyCourse();
+        Course course2 = database.addEmptyCourse();
+
+        ModelingExercise modelingExercise = ModelFactory.generateModelingExercise(now.minusDays(1), now.minusHours(2), now.minusHours(1), DiagramType.ClassDiagram, course1);
+        modelingExercise = modelingExerciseRepository.save(modelingExercise);
+        database.addGradingInstructionsToExercise(modelingExercise);
+
+        // Create example submission
+        var exampleSubmission = database.generateExampleSubmission("model", modelingExercise, true);
+        exampleSubmission = database.addExampleSubmission(exampleSubmission);
+        database.addResultToSubmission(exampleSubmission.getSubmission(), AssessmentType.MANUAL);
+        var submission = submissionRepository.findWithEagerResultAndFeedbackById(exampleSubmission.getSubmission().getId()).get();
+        database.addFeedbackToResult(ModelFactory.generateFeedback().stream().findFirst().get(), Objects.requireNonNull(submission.getLatestResult()));
+
+        modelingExercise.setCourse(course2);
+        request.postWithResponseBody("/api/modeling-exercises/import/" + modelingExercise.getId(), modelingExercise, ModelingExercise.class, HttpStatus.CREATED);
```

Comment:
should you maybe also test the behavior of the call more, maybe check that the response is correct? Or is this already done in another test?

Reply:
Thanks for the heads-up. I added some assertions.